### DISCUSSION TITLE
How to wait for file is actually stored

### DISCUSSION THREAD

**Discussion by polar-webspark** [spatie/laravel-medialibrary#2901] [Q&A]
I save my .kmz file like this
```
public function saveKmz(UploadedFile $kmz, string $locale = 'en'): Media
    {
        $this->clearMediaCollection('kmz_' . $locale);

        return $this->addMedia($kmz)
            ->withCustomProperties(['root' => 'stageKmz/' . $this->getId()])
            ->toMediaCollection('kmz_' . $locale, 'public');
    }
```
Then i trigger event and try to unzip my file in listener. I got **file not exists** error. But when i look in the directory, my file is rigth here. So i think its just didnt wait for file is actually stored and goes on. How can i fix this?

**Comment by angeljqv** [comment]
>How can i fix this?

Find a way and tell us
